src/entries/smt.rs

```diff
@@ -0,0 +1,188 @@
+use crate::entries::helper::{
+    generate_claim_key, generate_claim_value, generate_cota_index, generate_define_key,
+    generate_define_value, generate_hold_key, generate_hold_value, generate_withdrawal_key,
+    generate_withdrawal_key_v1, generate_withdrawal_value, generate_withdrawal_value_v1,
+};
+use crate::indexer::index::get_cota_smt_root;
+use crate::models::claim::ClaimDb;
+use crate::models::common::get_all_cota_by_lock_hash;
+use crate::models::define::DefineDb;
+use crate::models::hold::HoldDb;
+use crate::models::withdrawal::WithdrawDb;
+use crate::smt::db::cota_db::CotaRocksDB;
+use crate::smt::db::schema::{
+    COLUMN_SMT_BRANCH, COLUMN_SMT_LEAF, COLUMN_SMT_ROOT, COLUMN_SMT_TEMP_LEAVES,
+};
+use crate::smt::store::smt_store::SMTStore;
+use crate::smt::CotaSMT;
+use crate::utils::error::Error;
+use crate::utils::helper::diff_time;
+use chrono::prelude::*;
+use cota_smt::common::*;
+use cota_smt::smt::{blake2b_256, H256};
+use log::debug;
+use std::collections::HashMap;
+
+pub async fn generate_history_smt<'a>(
+    db: &'a CotaRocksDB,
+    lock_script: Vec<u8>,
```

Comment:
I think we should use `&[u8]` or `AsRef<[u8]>` as the `lock_script` here, which can reduce the vec clone since in the following code we only need to use the slice calculation.